django/loader/reference_binder.py

```diff
@@ -92,23 +92,31 @@ def bind(ref, path):
                 source_ref = (fhir_object["resourceType"], path)
                 self.cache.sadd(target_ref, json.dumps((source_ref, fhir_object["id"])))
 
-        def rec_bind_existing_reference(fhir_object, reference_path, sub_path=""):
-            sub_fhir_object = fhir_object[reference_path[0]]
-            if sub_path:
-                sub_path = f"{sub_path}."
-            if len(reference_path) == 1:
-                # If we have a list of references, we want to bind all of them.
-                # Thus, we loop on all the items in reference_attribute.
+        def rec_bind_existing_reference(fhir_object, reference_path: List[str], sub_path=""):
```

Comment:
I'm pretty sure we can get rid of this function and let `bind_existing_reference` itself be recursive